src/codeu/chat/client/commandline/Panel.java

```diff
@@ -17,42 +17,394 @@
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Scanner;
+import java.util.Stack;
+import java.util.ArrayList;
+import java.util.List;
+import java.io.IOException;
 
-// PANEL
-//
-// A panel is a collection of commands that are to be executed within a specifc
-// context. Similar to how panels are used in a GUI, is is a command line
-// equivalent.
-//
-final class Panel {
-
-  public interface Command {
-    void invoke(Scanner line);
-  }
+import codeu.chat.client.core.Context;
+import codeu.chat.client.core.ConversationContext;
+import codeu.chat.client.core.MessageContext;
+import codeu.chat.client.core.UserContext;
+import codeu.chat.util.Tokenizer;
 
-  private final Map<String, Command> commands = new HashMap<>();
+public final class Chat {
```

Comment:
I think I understand what was going on here. After I finished my activity, I tried my hand at Version Check, which used Chat and had me changing it in ways I didn't with Tokenization. I wanted to revert those changes as soon as the repo was up. A copy of the original Chat was on my clipboard, so I was able to select and paste over it for the current version of chat, but I must've pasted it here too by accident, sorry.